live-build/misc/upgrade-scripts/verify-impl

```diff
@@ -44,6 +44,81 @@ function cleanup() {
 	return "$rc"
 }
 
+function verify_run_verify_jar() {
+	echo "Verify - Run verify jar."
+
+	/usr/bin/java \
+		-Dlog.dir=/var/delphix/server/upgrade-verify \
+		-Dmdsverify=true \
+		"$VERIFY_DEBUG_OPT" \
+		-jar /opt/delphix/server/lib/exec/upgrade-verify/upgrade-verify.jar \
+		-d "${opt_o:-$DROPBOX_DIR/upgrade_verify_report.json}" \
+		-f "${opt_f:-1}" \
+		-l "${opt_l:-en-US}" \
+		-v "$IMAGE_VERSION" \
+		-pl 60 -ph 80 \
+		"$VERIFY_LIVE_MDS_OPT" ||
+		die "'upgrade-verify.jar' failed in verification container"
+
+	#
+	# This name is used by the "upgrade-verify.jar" just executed, so we
+	# cannot change this value without also modifying that JAR.
+	#
+	MDS_SNAPNAME="MDS-CLONE-upgradeverify"
+
+	/opt/delphix/server/bin/dx_manage_pg stop -s "$MDS_SNAPNAME" ||
+		die "failed to stop postgres"
+
+	/opt/delphix/server/bin/dx_manage_pg cleanup -s "$MDS_SNAPNAME" ||
+		die "failed to cleanup postgres"
+}
+
+function verify_stack_startup() {
+	echo "Verify - Run stack startup."
+
+	#
+	# Make sure delphix-mgmt and delphix-postgres aren't running within container.
+	#
+	[[ $(systemctl is-active delphix-mgmt) == 'inactive' ]] ||
+		die "delphix-mgmt service is not inactive."
+	[[ $(systemctl is-active delphix-postgres@default.service) == 'inactive' ]] ||
+		die "delphix-postgres@default.service service is not inactive."
+
+	#
+	# Edit Postgres systemd.service file to run within container.
+	#
+	local delphix_postgres_service='/lib/systemd/system/delphix-postgres@.service'
+	sed -i 's/ConditionVirtualization/# &/' ${delphix_postgres_service}
+	systemctl daemon-reload
+
+	#
+	# We need to make a copy of MDS within container to avoid writing data to the host MDS.
```

Comment:
Why not create a clone of MDS, and use the clone, similar to what the upgrade-verify JAR does? Would that work? A clone would likely be more efficient to create, and clean up.

Reply:
Good call. I've made change to leverage `dx_manage_pg.sh` script to start and stop a copy of MDS. The script creates a ZFS of snapshot and uses that to start the instance internally.